### ISSUE TITLE
git history really big, consider making new repos for each semester

### ISSUE THREAD

**Issue by SamLau95** [data-8/data8assets#2]
Looks like the `.git` folder is really big:

```
$ du -sh data8assets/.git                                                                                  
1.1G    data8assets/.git
```

This is probably because git keeps the history of adding / removing big datasets.

Students had trouble downloading their work since `tar` would zip the `.git` folder too. For now I've told them to run `tar` on the `data8assets/materials` folder instead (~150 MB).

Since `git clone` causes a full download of the history, next semester we'll have 500 students trying to download a >1 GB repo which seems wasteful.

We can avoid this by making new repos for each semester (`materials_sp17`, `materials_fa17`, etc), or by squashing the history for this repo. I prefer the first option since it avoids rewriting `git` history which makes `git pull` more complicated for the staff.

**Comment by yuvipanda** [comment]
We totally should do this. We could archive this repo, and then just have a new repo here that just has squashed history. Or just have a specific repo per semester as @SamLau95 mentioned. I think having a full checkout here for each student is going to cause problems with the fall infrastructure.

@SamLau95 @papajohn objections to doing this?

**Comment by SamLau95** [comment]
@yuvipanda Sounds like a great idea!

**Comment by papajohn** [comment]
@vinitra

**Comment by vinitra-zz** [comment]
On it! @papajohn

**Comment by papajohn** [comment]
Fine with me. Let's make sure all the old stuff stays published and that
all links are permanent.

**Comment by yuvipanda** [comment]
Thanks @vinitra. I see the new materials repo, and it's at 90M - which is much better! It looks like you deleted old files as a commit - https://github.com/data-8/materials-fa17/commit/ae3fcd3a337e96d9d6b4af67164630db1672b915. Since git keeps all history, deleting like this doesn't actually reduce the size of the repo! You should use something like https://help.github.com/articles/removing-sensitive-data-from-a-repository/. Let's do that before classes start - am happy to spend some time with you to get it done.

**Comment by vinitra-zz** [comment]
@yuvipanda thanks for the insight! I think I fixed it... can you confirm?

**Comment by yuvipanda** [comment]
Nope, it's still there: https://github.com/data-8/materials-fa17/commit/51bbdfd1dec1535cb7ffc547274425d8b2edb08e

**Comment by yuvipanda** [comment]
@vinitra I've done some force pushing and cleaned it out. People who had clones locally might have to re-clone - let me know if people run into issues and I can help.
